tests/test_pipelines_common.py

```diff
@@ -337,14 +337,16 @@ def test_float16_inference(self):
         pipe.to(torch_device)
         pipe.set_progress_bar_config(disable=None)
 
+        output = pipe(**self.get_dummy_inputs(torch_device))[0]
+        
+        components = self.get_dummy_components()
         for name, module in components.items():
             if hasattr(module, "half"):
                 components[name] = module.half()
         pipe_fp16 = self.pipeline_class(**components)
         pipe_fp16.to(torch_device)
         pipe_fp16.set_progress_bar_config(disable=None)
 
-        output = pipe(**self.get_dummy_inputs(torch_device))[0]
```

Comment:
In this pipeline's `__init__` , it creates an `attention_store` object and add to both the pipeline and all the cross-attention processors in unet.

here we used the same unet to create 2 pipelines, so when we create the second pipeline, it changed the `attention_store` object in the the first pipeline's unet - so now the `attention_store` in the first pipeline and its unet are pointing to different things and this cause the test to fail

I changed the test here a little bit to make it work but also think the design of `attention_store` might be a little bit flawed?

Reply:
Hmm, but this also means we have a problem if the pipeline is called twice in a row no? Think the code should stay as is here and we make sure that before every forward pass all the `attention_store`s are freed

Reply:
ohh yeah maybe we can add the `attention_store` to unet during `forward`, i.e.  move the `self.register_attention_control()` from the `__init__` to `__call__` - I will test it out